pkg/controller/keyvault/secret/keyvault_secret.go

```diff
@@ -0,0 +1,192 @@
+	}
+
+	lateInit := false
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	secretclients.LateInitialize(&cr.Spec.ForProvider, secret)
+	if !cmp.Equal(currentSpec, &cr.Spec.ForProvider) {
+		if err := c.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errUpdateSecretCRFailed)
+		}
+		lateInit = true
+	}
+
+	cr.Status.SetConditions(xpv1.Available())
+	cr.Status.AtProvider = secretclients.GenerateObservation(secret)
+
+	isUpToDate, err := secretclients.IsUpToDate(ctx, c.kube, cr.Spec.ForProvider, &secret)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errCheckUpToDate)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:          true,
+		ResourceUpToDate:        isUpToDate,
+		ResourceLateInitialized: lateInit,
+	}, nil
+}
+
+func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.KeyVaultSecret)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotSecret)
+	}
+
+	cr.Status.SetConditions(xpv1.Creating())
+
+	val, err := secretclients.ExtractSecretValue(ctx, c.kube, &cr.Spec.ForProvider.Value)
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+
+	_, err = c.client.SetSecret(ctx, cr.Spec.ForProvider.VaultBaseURL, cr.Spec.ForProvider.Name, keyvault.SecretSetParameters{
+		Value:            azure.ToStringPtr(val),
+		Tags:             azure.ToStringPtrMap(cr.Spec.ForProvider.Tags),
+		ContentType:      cr.Spec.ForProvider.ContentType,
+		SecretAttributes: secretclients.GenerateAttributes(cr.Spec.ForProvider.SecretAttributes),
+	})
+
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateFailed)
+	}
+
+	return managed.ExternalCreation{}, nil
+}
+
+func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.KeyVaultSecret)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotSecret)
+	}
+
+	val, err := secretclients.ExtractSecretValue(ctx, c.kube, &cr.Spec.ForProvider.Value)
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	_, err = c.client.SetSecret(ctx, cr.Spec.ForProvider.VaultBaseURL, cr.Spec.ForProvider.Name, keyvault.SecretSetParameters{
+		Value:            azure.ToStringPtr(val),
+		Tags:             azure.ToStringPtrMap(cr.Spec.ForProvider.Tags),
+		ContentType:      cr.Spec.ForProvider.ContentType,
+		SecretAttributes: secretclients.GenerateAttributes(cr.Spec.ForProvider.SecretAttributes),
+	})
+
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdateFailed)
+	}
+
+	return managed.ExternalUpdate{}, nil
+}
+
+func (c *external) Delete(ctx context.Context, mg resource.Managed) error {
+	cr, ok := mg.(*v1alpha1.KeyVaultSecret)
+	if !ok {
+		return errors.New(errNotSecret)
+	}
+	cr.Status.SetConditions(xpv1.Deleting())
```

Comment:
crossplane-runtime can now [handle setting "creating" and "deleting" conditions](https://github.com/crossplane/crossplane-runtime/pull/242).